codex/sales/states/proving.nim

```diff
@@ -129,27 +130,35 @@ method run*(state: SaleProving, machine: Machine): Future[?State] {.async.} =
   without clock =? context.clock:
     raiseAssert("clock not set")
 
-  debug "Start proving", requestId = data.requestId, slotIndex = data.slotIndex
   try:
-    let loop = state.proveLoop(market, clock, request, data.slotIndex, onProve)
-    state.loop = loop
-    await loop
-  except CancelledError:
-    discard
+    debug "Start proving", requestId = data.requestId, slotIndex = data.slotIndex
+    try:
+      let loop = state.proveLoop(market, clock, request, data.slotIndex, onProve)
+      state.loop = loop
+      await loop
+    except CancelledError as e:
+      trace "proving loop cancelled"
+      discard
+    except CatchableError as e:
+      error "Proving failed",
+        msg = e.msg, typ = $(type e), stack = e.getStackTrace(), error = e.msgDetail
+      return some State(SaleErrored(error: e))
+    finally:
+      # Cleanup of the proving loop
+      debug "Stopping proving.", requestId = data.requestId, slotIndex = data.slotIndex
+
+      if not state.loop.isNil:
+        if not state.loop.finished:
+          try:
+            await state.loop.cancelAndWait()
+          except CatchableError as e:
+            error "Error during cancellation of proving loop", msg = e.msg
+
+        state.loop = nil
+
+    return some State(SalePayout())
+  except CancelledError as e:
+    trace "SalePreparing.run onCleanUp was cancelled", error = e.msgDetail
```

Comment:
```suggestion
    trace "SaleProving.run onCleanup was cancelled", error = e.msgDetail
```

Same below

Reply:
Yikes, sorry mark, I missed this one. IMO, `onCleanup` should be removed entirely from that trace log. Can do in a future PR.